swift-sdk/Internal/RequestCreator.swift

```diff
@@ -16,27 +16,31 @@ struct RequestCreator {
     // MARK: - API REQUEST CALLS
     
     func createUpdateEmailRequest(newEmail: String) -> Result<IterableRequest, IterableError> {
-        var body: [String: Any] = [JsonKey.newEmail: newEmail]
+        if case .none = auth.emailOrUserId {
+            ITBError(authFailedMessage)
+            return .failure(IterableError.general(description: authFailedMessage))
+        }
+        
+        var body = [String: Any]()
         
         if let email = auth.email {
             body[JsonKey.currentEmail] = email
         } else if let userId = auth.userId {
             body[JsonKey.currentUserId] = userId
-        } else {
-            ITBError("Both email and userId are nil")
-            return .failure(IterableError.general(description: "Both email and userId are nil"))
         }
         
+        body[JsonKey.newEmail] = newEmail
+        
         return .success(.post(createPostRequest(path: Const.Path.updateEmail, body: body)))
     }
     
     func createRegisterTokenRequest(registerTokenInfo: RegisterTokenInfo,
                                     notificationsEnabled: Bool) -> Result<IterableRequest, IterableError> {
         if case .none = auth.emailOrUserId {
-            ITBError("Both email and userId are nil")
-            return .failure(IterableError.general(description: "Both email and userId are nil"))
+            ITBError(authFailedMessage)
```

Comment:
'authFailedMessage' is a generic name for variable. This is a very specific one so we should name it differently. 'authMissing', 'userCredentialMissing'  or of that kind.